Registration/Authorization

Nearzev opened the pull request:
feat: добавлена базовая верстка, валидация формы и стили

Nearzev committed:
feat: созданы компоненты для формы и инпутов и стили к ним

Nearzev committed:
feat: добавлена валидация форм, внесены небольшие изменения в стили

dimonarhipon committed:
Merge branch 'develop' into registrationa/authorization

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+@use '../../shared/styles/vars' as *;
```
Интересная реализация. По-моему путь '../../shared' можно заменить на '@/shared'

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+    align-items: center;
+    justify-content: center;
+
+    width: 471px;
```
Да, в макете 471px. Это ошибка дизайнера, думаю он хотела 470.
Фиксированную ширину проставлять не рекомендуется, т.к. в mobile разрешении будет горизонтальный скрол.
Тут можно max-width: 470px; width: 100%; и не нужно прописывать стили для mobile разрешений

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+    width: 471px;
+    border-radius: 20px;
+
+    background-color: #fff;
```
Тут можно использовать переменную $white

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+    font-family: Overpass;
+    font-size: 20px;
+    font-weight: 400;
+    letter-spacing: 0em;
```
Согласно кодексу 2.9. в проекте используется px

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+    padding: 45px 60px;
+}
+
+.fromInput {
```
наверно хотел formInput
В целом по неймингу классов, можно везде убрать слово form, т.к. оно излишне уточняет

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+    font-family: Overpass;
+    font-size: 20px;
+    font-weight: 400;
+    letter-spacing: 0em;
```
Согласно кодексу 2.9. в проекте используется px

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+
+.validErorrSpan {
+    color: red;
+    text-align: center;
```
Стандартно слева текст. По центру, на твоё авторскую реализацию

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+}
+
+.validErorrSpan {
+    color: red;
```
$pink тут плохо читается? Я не смотрел реализацию. Дизайнер чуть позже пропишет цвет.
Если вводится новый цвет, то нужно его выносить в vars.

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+    padding: 15px;
+
+    width: 350px;
+    height: 60px;
```
Высоту в input можно реализовать через padding.
Над шириной можно подумать, на мобилке всё равно придётся делать ширину адаптивной.
Тут max-width + width:100% не очень подходят (input на них плохо реагирует), поэтому нужно использовать flex-grow: 1

dimonarhipon commented on src/pages/AuthorizationPage/FormInput.tsx:
```diff
@@ -0,0 +1,30 @@
+
+    // const handleFormChange = (e:React.ChangeEvent<HTMLInputElement>) => {
+    //     props.setInputParam(e.target.value);
+    //     props.isValidForm(e.target.value, validPattern[props.name])
```
Согласно кодексу 3.7. в проекте не используется однобуквенные названия переменных.
Нужно e заменить на event

dimonarhipon commented on src/pages/AuthorizationPage/FormInput.tsx:
```diff
@@ -0,0 +1,30 @@
+
+const FormInput = (props: inputFormTypes) => {
+
+    // const handleFormChange = (e:React.ChangeEvent<HTMLInputElement>) => {
```
Согласно кодексу 3.12.1. в проекте другой нейминг обработчиков событий, а именно onFormChange
и по 3.15. такой комментарий на review не отправляем

dimonarhipon commented on src/pages/AuthorizationPage/FormInput.tsx:
```diff
@@ -0,0 +1,30 @@
+                type={props.type} 
+                placeholder={props.placeholder}
+                onChange={(e) => props.setInputParam(e.target.value)}
+                onBlur={(e:React.ChangeEvent<HTMLInputElement>) => props.isValidForm(e.target.value, validPattern[props.name], props.name)}
```
Допустимо, но плохо читаемо

dimonarhipon commented on src/pages/AuthorizationPage/Registration.module.scss:
```diff
@@ -0,0 +1,59 @@
+@use '../../shared/styles/vars' as *;
+
```
Некорректный нейминг для файла стилей. Он должен совпадать с названием компонента.

dimonarhipon commented on src/pages/AuthorizationPage/AuthorizationPage.tsx:
```diff
@@ -0,0 +1,44 @@
+                    isValidForm={isValidForm}
+                />
+                {!validForm.password && <span className={styles.validErorrSpan}>Пароль должен содержать латинские символы и быть не короче 5 символов</span> }
+                <button className={styles.formButton} type='button'>Регистрация</button>
```
тут спорно type="button"

dimonarhipon commented on src/pages/AuthorizationPage/AuthorizationPage.tsx:
```diff
@@ -0,0 +1,44 @@
+                    setInputParam={setPassword}
+                    isValidForm={isValidForm}
+                />
+                {!validForm.password && <span className={styles.validErorrSpan}>Пароль должен содержать латинские символы и быть не короче 5 символов</span> }
```
именуя класс css validErorrSpan идёт привязка к тегу.
И да, span тут семантически не подходит. Тут подходит p

через && - это правильно

dimonarhipon commented on src/pages/AuthorizationPage/AuthorizationPage.tsx:
```diff
@@ -0,0 +1,44 @@
+                    isValidForm={isValidForm}
+                />
+                {!validForm.username && <span className={styles.validErorrSpan}>Логин должен содержать только латинские символы и быть не короче 3 символов</span>  }
+                <FormInput 
```
Нужно добавить иконку скрыть/показать пароль

dimonarhipon commented on src/pages/AuthorizationPage/AuthorizationPage.tsx:
```diff
@@ -0,0 +1,44 @@
+                    setInputParam={setUsername}
+                    isValidForm={isValidForm}
+                />
+                {!validForm.username && <span className={styles.validErorrSpan}>Логин должен содержать только латинские символы и быть не короче 3 символов</span>  }
```
Согласно кодексу 3.8. можно вынести текст в переменную. И числа в константу.

dimonarhipon reviewed (commented):
Меня смущает что файл компонента лежит в pages. Следует вынести в отдельную папку widget.
Критичные ошибки те, которые не соответствуют кодексу.

Nearzev committed:
fix: исправлены ошибки, добавлена кнопка для просмотра пароля

1) Внесены исправления в стили и названия классов
2)Изменено расположение файлов
3)Добавлена функция просмотра вводимого пароля

Nearzev committed:
Merge remote-tracking branch 'origin/registrationa/authorization' into registrationa/authorization

Nearzev committed:
fix: перемещение картинки в папку public

dimonarhipon commented on src/Widgets/AuthorizationWidget/AuthorizationWidget.module.scss:
```diff
@@ -53,7 +54,30 @@
+    margin: 10px 0px;
+
+    width: 350px;
+    height: 60px;
```
Писал в прошлом review об этой ошибке

dimonarhipon commented on src/Widgets/AuthorizationWidget/AuthorizationWidget.module.scss:
```diff
@@ -32,14 +33,14 @@
-    width: 350px;
+    max-width: 350px;
+    width: 100%;
     height: 60px;
```
высоту кнопки корректнее реализовать через padding

dimonarhipon commented on src/Widgets/AuthorizationWidget/AuthorizationWidget.tsx:
```diff
@@ -0,0 +1,59 @@
+    const isValidForm = (formValue: string, validPattern: RegExp, formName: string) => {
+
+        validPattern.test(formValue) ? setValidForm({ ...validForm, [formName]: true }) : setValidForm({ ...validForm, [formName]: false });
+        console.log(validForm);
```
Консоль в PR отдавать нежелательно

dimonarhipon commented on src/Widgets/AuthorizationWidget/AuthorizationWidget.tsx:
```diff
@@ -0,0 +1,59 @@
+            setType('text');
+        } else {
+            setType('password');
+        }
```
Есть более красивая реализация
если type === text return setType(test)
return setType(password)

dimonarhipon commented on src/Widgets/AuthorizationWidget/AuthorizationWidget.tsx:
```diff
@@ -0,0 +1,59 @@
+                <FormInput
+                    name='password'
+                    type={type}
+                    placeholder='Пароль'
```
Согласно кодексу 3.8. строки не используем там, где можно использовать переменную

dimonarhipon commented on src/Widgets/AuthorizationWidget/AuthorizationWidget.tsx:
```diff
@@ -0,0 +1,59 @@
+                <FormInput
+                    name='username'
+                    type='text'
+                    placeholder='Логин'
```
Кодекс 3.8.

dimonarhipon commented on src/Widgets/AuthorizationWidget/FormInput.tsx:
```diff
@@ -0,0 +1,27 @@
+                onChange={(event) => props.setInputParam(event.target.value)}
+                onBlur={(event: React.ChangeEvent<HTMLInputElement>) => props.isValidForm(event.target.value, validPattern[props.name], props.name)}
+            />
+            {props.name === 'password' && <div onClick={props.passwordHide} className={styles.passwordHide}><img src='./eye.svg' alt='eyePassword' /></div>}
```
интерактивную иконку можно вставить инлайново, не импортируя

dimonarhipon commented on src/Widgets/AuthorizationWidget/AuthorizationWidget.module.scss:
```diff
@@ -1,4 +1,4 @@
-@use '../../shared/styles/vars' as *;
+@use '@/shared/styles/vars' as *;
 
```
Зачем второе слово в компоненте Page/Widget? Так идёт жёсткая привязка к папке. Предстать, что через 2 недели тебе надо переместить компонент в папку features.

dimonarhipon reviewed (commented):
Иконок/изображения должны лежать в отдельной папке. В макете ещё есть одна иконка для пароля.

Nearzev committed:
fix: внесены исправления добавлена новая иконка

dimonarhipon commented on src/Widgets/Authorization/Authorization.tsx:
```diff
@@ -10,6 +10,22 @@ export const AuthorizationPage = () => {
         username: true,
         password: true,
     });
+    const forms = {
```
Пространство имён можно хранить вне компонента. Т.к. сейчас при каждом render параметры будут каждый раз создаваться.

dimonarhipon commented on src/Widgets/Authorization/Authorization.tsx:
```diff
@@ -19,39 +35,39 @@ export const AuthorizationPage = () => {
+        type === 'password' ? setType('text') : setType('password');
+    };
+
+    const buttonIsDisabled = () => {
```
Согласно кодексу 3.12.2. is нужно ставить вначале функции
